service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureVmStep.java

```diff
@@ -91,15 +91,27 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
               .getByResourceGroup(
                   azureCloudContext.getAzureResourceGroupId(), networkResource.getNetworkName());
 
+      var createNic =
+          computeManager
+              .networkManager()
+              .networkInterfaces()
+              .define(String.format("nic-%s", resource.getVmName()))
+              .withRegion(resource.getRegion())
+              .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+              .withExistingPrimaryNetwork(existingNetwork)
+              .withSubnet(networkResource.getSubnetName())
+              .withPrimaryPrivateIPAddressDynamic()
+              .withExistingPrimaryPublicIPAddress(
+                  existingAzureIp) // TODO this needs to be updated to support not exposing public
```

Comment:
maybe put a ticket number here for this work

Reply:
yea...it's https://broadworkbench.atlassian.net/browse/TOAZ-79...I'm going to leave things as is since the ticket is already assigned and we're expecting it to be done soon